utils/sql/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -783,65 +895,81 @@ public ArtifactSearchResults searchArtifacts(String search, int offset, int limi
                                 + "v.description LIKE ? OR "
                                 + "EXISTS(SELECT l.globalId FROM labels l WHERE l.label = ? AND l.globalId = v.globalId)"
                                 + ")");
                         binders.add((query, idx) -> {
-                            query.bind(idx, "%" + search + "%");
+                            query.bind(idx, "%" + filter.getValue() + "%");
+                        });
+                        binders.add((query, idx) -> {
+                            query.bind(idx, "%" + filter.getValue() + "%");
                         });
                         binders.add((query, idx) -> {
-                            query.bind(idx, "%" + search + "%");
+                            query.bind(idx, "%" + filter.getValue() + "%");
                         });
                         binders.add((query, idx) -> {
-                            query.bind(idx, "%" + search + "%");
+                            query.bind(idx, "%" + filter.getValue() + "%");
                         });
                         binders.add((query, idx) -> {
-                            // Note: convert search to lowercase when searching for labels (case-insensitivity support).
-                            query.bind(idx, search.toLowerCase());
+                          //    Note: convert search to lowercase when searching for labels (case-insensitivity support).
+                            query.bind(idx, filter.getValue().toLowerCase());
                         });
                         break;
                     case labels:
                         where.append("EXISTS(SELECT l.globalId FROM labels l WHERE l.label = ? AND l.globalId = v.globalId)");
                         binders.add((query, idx) -> {
-                            // Note: convert search to lowercase when searching for labels (case-insensitivity support).
-                            query.bind(idx, search.toLowerCase());
+                          //    Note: convert search to lowercase when searching for labels (case-insensitivity support).
+                            query.bind(idx, filter.getValue().toLowerCase());
                         });
                         break;
                     case name:
                         where.append("(v.name LIKE ?) OR (a.artifactId LIKE ?)");
                         binders.add((query, idx) -> {
-                            query.bind(idx, "%" + search + "%");
+                            query.bind(idx, "%" + filter.getValue() + "%");
                         });
                         binders.add((query, idx) -> {
-                            query.bind(idx, "%" + search + "%");
+                            query.bind(idx, "%" + filter.getValue() + "%");
                         });
                         break;
+                    case group:
+                        where.append("(v.groupId LIKE ?)");
```

Comment:
This should be an equality search instead of a `LIKE` - just like filtering by tags, filtering by group is exact.

Reply:
I think you say this because we are using the search method in the rest layer for `listArtifactsInGroup`.

I agree `listArtifactsInGroup` should filter by exact groupId, but for the search api it's nice to not be exact.

So I think we should have to add a method to the storage layer for listing the artifacts in a group

Reply:
I don't have a strong opinion on this, but I do have a slight preference for making search-by-group strict.  I'd be interested in the opinions of the rest of @Apicurio/developers though!

Reply:
what about adding to the `SearchFilter` a property that indicates if the search is by exact text or a  `LIKE` query?

Reply:
My opinion depends on how are we going to implement this in the UI. If we're going to add a drop-down and let the user select one group, then +1 to the equality filter. If the search field will be a text box, then +1 to the like approach. Fabian's suggestion about using an additional property also seems reasonable.

Reply:
At least temporarily, the UI will be a text box, not a dropdown.  But you will be able to click on a Group to view the artifacts in that group.  Clicking a group will simply add the "group=xyz" filter to the query (it's not a separate page).  But still, as a user I wouldn't want to see other artifacts when I do that.

**For now** let's change this to be `=` instead of `LIKE` and also add a GitHub issue to follow up on this later.  I expect some changes to the UI which could influence (or be influenced) by this discussion.  So let's push this off to the future a bit.

Reply:
ok

Reply:
just out of curiosity @EricWittmann , how is the UI going to get the list of groups? we don't have that available in the API

Reply:
The UI won't get the list of groups.  But there will be links to a single Group from a couple of places:

1) From the list of Artifacts - each artifact in the list will also display its group, which will be clickable
2) From the details of an Artifact, we will show the Group in the breadcrumbs, which will be clickable